### PR TITLE
Upload api groups

### PR DESCRIPTION
**Pull request by SteveRedka**

### CODE REVIEW

File: lib/uploadcare/client/group_client.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  # Groups serve a purpose of better organizing files in your Uploadcare projects.
+  # You can create one from a set of files by using their UUIDs.
+  # https://uploadcare.com/docs/api_reference/upload/groups/
+  class GroupClient < ApiStruct::Client
+    upload_api
+
+    # Create files group from a set of files by using their UUIDs.
+    # https://uploadcare.com/api-refs/upload-api/#operation/createFilesGroup
+
+    def create(file_ids, **options)
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/group_client.rb]: `POST /group/` accepts not only UUIDs, but full CDN URLs as well (e.g. to add effects/crops/resizes).

File: lib/uploadcare/client/rest_group_client.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal true
+
+module Uploadcare
+  # https://uploadcare.com/api-refs/rest-api/v0.5.0/#tag/Group/paths/~1groups~1%3Cuuid%3E~1storage~1/put
+  class RestGroupClient < ApiStruct::Client
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/rest_group_client.rb]: don't forget to add RestClient here

File: lib/uploadcare/client/upload_client.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    def files_formdata(arr)
+      arr.map { |file| [HTTP::FormData::File.new(file).filename,
+                        HTTP::FormData::File.new(file)] }.to_h
+    end
+
+    def file?(object)
+      object.respond_to?(:path) && ::File.exist?(object.path)
+    end
+  end
+end
```
Comment (dmitry-mukhin) [inline lib/uploadcare/client/upload_client.rb]: this probably should be in other PR?
can we ignore this here?

File: lib/uploadcare/service/upload/signature_generator.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        to_sign = SECRET_KEY + expires_at.to_s
+        signature = Digest::MD5.hexdigest(to_sign)
+        {
+          'signature': signature,
+          'expire': expires_at
+        }
+      end
+    end
+  end
+end
```
Comment (dmitry-mukhin) [inline lib/uploadcare/service/upload/signature_generator.rb]: this probably should be in other PR

File: lib/uploadcare/client/group_client.rb (hunk 1 above)
Comment (SteveRedka) [reply lib/uploadcare/client/group_client.rb]: Is this a problem?

It accepts an array of strings and passes them to API without modification

Comment (dmitry-mukhin) [reply lib/uploadcare/client/group_client.rb]: it's misleading.
simply rename argument to something else. e.g. `file_list`

**Pull request closed without merging**